tests/test_app.py

```diff
@@ -225,13 +240,99 @@ def test_address_search_with_page_calls_api_correctly(self, mock_get):
         mock_get.assert_called_with('http://landregistry.local:8004/title_search_address/PLYMOUTH',
                                     params={'page': 23})
 
-    @mock.patch('requests.get', return_value=unavailable_title)
-    def test_get_title_page_no_title(self, mock_get):
-        mock_get.side_effect = Exception('problem!')
-        response = self.app.get('/titles/titleref')
-        assert response.status_code == 500
-        assert 'Sorry, we are experiencing technical difficulties.' in response.data.decode()
+    @mock.patch('requests.get', return_value=fake_postcode_search)
+    def test_search_title_passes_postcode_to_api_with_space_added_when_missing(self, mock_get):
+        search_term = 'PL98TB'
+        expected_postcode = 'PL9 8TB'
+        self.app.get('/title-search/{}'.format(search_term))
+
+        assert len(mock_get.mock_calls) == 1
+        actual_call = mock_get.mock_calls[0]
+        url_param = actual_call[1][0]
+        assert url_param.endswith('title_search_postcode/{}'.format(expected_postcode))
+
+    @mock.patch('requests.get', return_value=fake_postcode_search)
+    def test_search_title_passes_postcode_to_api_as_it_is_when_valid(self, mock_get):
+        search_term = 'PL9 8TB'
+        self.app.get('/title-search/{}'.format(search_term))
+
+        assert len(mock_get.mock_calls) == 1
+        actual_call = mock_get.mock_calls[0]
+        url_param = actual_call[1][0]
+        assert url_param.endswith('title_search_postcode/{}'.format(search_term))
+
+
+class TestAuthentication:
```

Comment:
I'm wrong! If it's not there, it tries to get it from the API. Can change it now :)